modules/programs/nushell.nix

```
@@ -177,39 +193,56 @@ in {
     };
   };
 
-  config = mkIf cfg.enable {
+  config = lib.mkIf cfg.enable {
     home.packages = [ cfg.package ];
 
-    home.file = mkMerge [
+    home.file = lib.mkMerge [
       (let
         writeConfig = cfg.configFile != null || cfg.extraConfig != ""
-          || aliasesStr != "";
+          || aliasesStr != "" || cfg.settings != { };
+
+        aliasesStr = lib.concatLines
+          (lib.mapAttrsToList (k: v: "alias ${k} = ${v}") cfg.shellAliases);
```

```suggestion
          (lib.mapAttrsToList (k: v: "alias '${k}' = ${v}") cfg.shellAliases);
```
This should make it possible to define aliases with spaces. I figured it might make sense to include in this PR, if that's alright.

---

Good call, I never thought about multi word aliases being valid. Added support in [cf27650](https://github.com/nix-community/home-manager/pull/6184/commits/cf27650610d793c4b17dbc207eac75d4e1feefc7)!

---

I used `toNushell` instead of manually adding quotes because quotes are valid parts of an alias name apparently. They don't work in some cases but that's a nushell issue, not a config issue
```nushell
# Works
alias 'a"a"a' = echo 'closed quotes work'
a"a"a

# Doesn't work
alias 'a"a' = echo 'echo opened quotes dont'

a"a
#   ┬
#   ╰── expected closing "
```

---

Awesome, thanks for including that!